src/privileged/autoplay/api.js

```diff
@@ -0,0 +1,397 @@
+"use strict";
+
+ChromeUtils.import("resource:///modules/PermissionUI.jsm");
+ChromeUtils.import("resource:///modules/SitePermissions.jsm");
+ChromeUtils.import("resource://gre/modules/ExtensionCommon.jsm");
+ChromeUtils.import("resource://gre/modules/ExtensionStorage.jsm");
+ChromeUtils.import("resource://gre/modules/PopupNotifications.jsm");
+ChromeUtils.import("resource://gre/modules/PrivateBrowsingUtils.jsm");
+ChromeUtils.import("resource://gre/modules/Services.jsm");
+ChromeUtils.import("resource://gre/modules/TelemetryController.jsm");
+
+const CID = ChromeUtils.import("resource://gre/modules/ClientID.jsm", {});
+const { AddonStudies } = ChromeUtils.import("resource://normandy/lib/AddonStudies.jsm", {});
+const { EventManager } = ExtensionCommon;
+
+XPCOMUtils.defineLazyGetter(this, "gBrowserBundle", function() {
+  return Services.strings.createBundle("chrome://browser/locale/browser.properties");
+});
+
+function _once(target, name) {
+  const p = new Promise(function(resolve, reject) {
+    target.addEventListener(name, function() {
+      resolve();
+    }, {once: true});
+  });
+  return p;
+}
+
+function getTelemetryId() {
+  const id = TelemetryController.clientID;
+  if (id === undefined) {
+    return CID.ClientIDImpl._doLoadClientID();
```

Comment:
Why not just `await ClientID.getClientID()`? It's already [doing this for you](https://searchfox.org/mozilla-central/rev/881a3c5664ede5e08ee986d76433bc5c4b5680e6/toolkit/modules/ClientID.jsm#173) and better to not call private methods in other modules :)

Reply:
OK! Thank you!

Reply:
Ah, I found that this function should be removed, since I have already made TelemetryController to generate `clientId`.